catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -147,13 +155,42 @@ public SourceResponse query(QueryRequest request) throws UnsupportedQueryExcepti
       query.setFacetMinCount(textFacetProp.getMinFacetCount());
     }
 
-    long totalHits;
-    List<Result> results = new ArrayList<>();
-    Map<String, Serializable> responseProps = new HashMap<>();
+    Serializable suggestQueryProp = request.getPropertyValue(SUGGESTION_REQUEST_KEY);
+
+    if (suggestQueryProp instanceof String) {
+      query = new SolrQuery();
+      query.setRequestHandler("/suggest");
+      query.setParam(SUGGEST_Q, (String) suggestQueryProp);
+      query.setParam(SUGGEST_CONTEXT_FILTER_QUERY, "gazetteer");
+    }
+
+    long totalHits = 0;
+
     try {
       QueryResponse solrResponse = client.query(query, SolrRequest.METHOD.POST);
-      totalHits = solrResponse.getResults().getNumFound();
+
       SolrDocumentList docs = solrResponse.getResults();
+      if (docs != null) {
+        totalHits = docs.getNumFound();
+        for (SolrDocument doc : docs) {
+          if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("SOLR DOC: {}", doc.getFieldValue(Metacard.ID + SchemaFields.TEXT_SUFFIX));
+          }
+          ResultImpl tmpResult;
+          try {
+            tmpResult = createResult(doc);
+          } catch (MetacardCreationException e) {
+            throw new UnsupportedQueryException("Could not create metacard(s).", e);
```

Comment:
✏️ `...result metacards(s)`?

Reply:
I think I just moved this code, but I like this suggestion. 👍